MakingItBetterG55/src/application/EventDisplayer.java

```diff
@@ -0,0 +1,235 @@
+package application;
+
+import javafx.geometry.Pos;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.BorderPane;
+import javafx.scene.text.Font;
+import javafx.scene.text.FontWeight;
+import javafx.stage.Modality;
+import javafx.stage.Stage;
+import javafx.stage.StageStyle;
+
+public class EventDisplayer {
+	private String title;
+	private String desc;
+	private String[] options;
+
+	// Feel free to change as much as this as you want, the primary goal was to serve as
+	// a graphical foundation for the event popup system.
+	public EventDisplayer(String title, String description, String[] options) {
+		this.title = title;
+		this.desc = splitDesc(description);
+		this.options = options;
+		openPopup(title, desc, options);
+	}
+
+	// Splitting up the description with a new line every 7 words
+	public String splitDesc(String description) {
+		String[] words = description.split(" ");
+		StringBuilder newDesc = new StringBuilder();
+
+		for (int i = 0; i < words.length; i++) {
+			newDesc.append(words[i]).append(" ");
+
+			if ((i + 1) % 7 == 0) {
+				newDesc.append("\n");
+			}
+		}
+		String updatedDesc = newDesc.toString();
+		return updatedDesc;
+	}
+
+	private void openPopup(String eventTitle, String eventDesc, String[] options) {
+		Stage eventStage = new Stage();
+		eventStage.initModality(Modality.APPLICATION_MODAL);
+		eventStage.initStyle(StageStyle.UNDECORATED);
+
+		AnchorPane eventLayout = new AnchorPane();
+		eventLayout.setStyle(
+				"-fx-border-color: #E1005C; -fx-border-width: 18px; -fx-background-color: #F79FC7; -fx-text-fill: white;");
+
+		Label titleLabel = new Label(eventTitle);
+		titleLabel.setStyle("-fx-text-fill: white;");
+		titleLabel.setFont(Font.font("SansSerif", FontWeight.BOLD, 40));
+		AnchorPane.setTopAnchor(titleLabel, 10.0);

```

Comment:
Why do you use  AnchorPane instead of eventLayout ( aka the instance of the AnchorPane class)?

Reply:
To my understanding, the AnchorPane class only lets you access the anchor constraint methods statically, so it would only let me use those methods through the class itself and not through an instance [eventLayout] of it. Here is the [Oracle Documentation](https://docs.oracle.com/javase/8/javafx/api/javafx/scene/layout/AnchorPane.html) for it if you're curious.

Reply:
Oh okay, coolio!